src/error.rs

```diff
@@ -37,6 +38,8 @@ pub enum Error {
     TooLarge,
     /// An invalid `Status`, such as `1337 ELITE`.
     Status,
+    /// An ivalid header value or header value fragment.
+    Invalid,
```

Comment:
Since most of the errors are about "invalid" values, I think this name should be specific to what kind of value was invalid. Perhaps `Header`.